### DISCUSSION TITLE
Local play for libVlc

### DISCUSSION THREAD

**Discussion by koying** [jellyfin/jellyfin-androidtv#846] [General]
Hi,

I have this setup where part of my library is in the cloud and part is local. 

Plex being no good for this setup, and Kodi being a mess, I have recently started looking at Jellyfin for its open-source approach. I thus created a single server in the cloud, that reaches to my local media by nfs+vpn.

Obviously, I'd like my local items to not go to cloud and back for playback so, for both libraries, I have "Shared folders" available, NFS for cloud and SMB for local, and that works fine in Kodi.

Now, I'd like  to use Jellyfin native app (i.e. this ;) ) to play my media, and it happens libVlc natively supports both NFS and SMB.
I have started looking at it and it seems I have a working POC.

Would there be any opposition to have a PR that enables "Local Play" to VLC, in addition to external player?

Would there be any knwown showstoppers I didn't stumble on, yet?

Thanks

**Comment by nielsvanvelzen** [comment]
Quoted from my reply in #845:

> The app only supports Jellyfin servers and will not support anything else.
> The app will always play the media via the server. The server can choose to read it from SMB or whatever but the client ALWAYS plays via the API of the server.

The app uses multiple player backends (right now ExoPlayer and LibVLC) so implementing such a feature should not be done on our side - not the player. Otherwise the behavior won't be consistent. But this adds a feature that not a lot of users would use but does need to be maintained, Especially with the current codebase this is a no-go.

**Reply by koying** [reply]
Thanks for the answer.
At least I know I don't have to keep my code tidy 😀
